lib/src/main/scala/code/chymyst/jc/Core.scala

```diff
@@ -7,10 +7,17 @@ import scala.collection.mutable
 
 object Core {
 
+  // List of molecules used as inputs by a reaction.
+  type InputMoleculeList = Array[(Molecule, AbsMolValue[_])]
```

Comment:
type name InputMolecule LIST (and comment LIST of molecules) for what is now an ARRAY!? Perhaps InputMolecules suffices (so you abstract out representation of Array (or List)).